autoprecompiles/src/constraint_optimizer.rs

```diff
@@ -110,7 +112,7 @@ fn solver_based_optimization<T: FieldElement, V: Clone + Ord + Hash + Display>(
             .map(|field| {
                 if let Some(n) = try_replace_by_number(field, solver) {
                     modified = true;
-                    new_algebraic_constraints.push(&n - field);
+                    new_algebraic_constraints.push(AlgebraicConstraint::assert_zero(&n - field));
```

Comment:
What about something like
```suggestion
                    new_algebraic_constraints.push(AlgebraicConstraint::assert_eq(n, field));
```
?

Reply:
sure